src/pyroscope-api-exporter.ts

```diff
@@ -86,7 +86,25 @@ export class PyroscopeApiExporter implements ProfileExporter {
 
     const formData: FormData = new FormData();
 
-    formData.append('profile', new Blob([profileBuffer]), 'profile');
+    // Convert Node Buffer -> *definitely* an ArrayBuffer for TS 5.9 DOM typings.
+    // Buffer.buffer is ArrayBufferLike (ArrayBuffer | SharedArrayBuffer).
+    // Narrow at runtime; if it's a SharedArrayBuffer, copy into a new ArrayBuffer.
+    let arrayBuffer: ArrayBuffer;
+    const { buffer, byteOffset, byteLength } = profileBuffer;
+    if (buffer instanceof ArrayBuffer) {
+      // Safe: returns ArrayBuffer
+      arrayBuffer = buffer.slice(byteOffset, byteOffset + byteLength);
+    } else {
+      // SharedArrayBuffer path: copy to a new ArrayBuffer
+      const copy = new Uint8Array(byteLength);
+      copy.set(profileBuffer);
+      arrayBuffer = copy.buffer; // ArrayBuffer
+    }
```

Comment:
I am thinking `ArrayBuffer.slice()` creates a copy, which may  be avoided.

```js
let a = new Uint8Array([1, 2, 3]);
let b = a.slice();
a[0] = 42;
console.log(a.buffer); // ArrayBuffer { [Uint8Contents]: <2a 02 03>, byteLength: 3 }
console.log(b.buffer); // ArrayBuffer { [Uint8Contents]: <01 02 03>, byteLength: 3 }
```

I think we can avoid a copy like this.
The first branch creates a reference  `Uint8Array<ArrayBuffer>` reference
The second branch creates `Uint8Array<SharedArrayBuffer>` and then slice returns a copy of type `Uint8Array<ArrayBuffer>`

```suggestion
    let arrayBuffer: Uint8Array<ArrayBuffer>;
    const { buffer, byteOffset, byteLength } = profileBuffer;
    if (buffer instanceof ArrayBuffer) {
      arrayBuffer = new Uint8Array(buffer, byteOffset, byteLength);
    } else {
      arrayBuffer = new Uint8Array(buffer, byteOffset, byteLength).slice();
    }
```

I think in practice, the buffer is never `SharedArrayBuffer`

Reply:
disclaimer: I am complete js noob, and I have not run this code, only compiled.